tests/lzma.rs

```diff
@@ -44,21 +60,30 @@ fn round_trip_with_options(
     info!("Compressed {} -> {} bytes", x.len(), compressed.len());
     #[cfg(feature = "enable_logging")]
     debug!("Compressed content: {:?}", compressed);
-    let mut bf = std::io::BufReader::new(compressed.as_slice());
-    let mut decomp: Vec<u8> = Vec::new();
-    lzma_rs::lzma_decompress_with_options(&mut bf, &mut decomp, decode_options)?;
-    Ok(decomp)
-}
 
-fn assert_round_trip_with_options(
-    x: &[u8],
-    encode_options: &lzma_rs::compress::Options,
-    decode_options: &lzma_rs::decompress::Options,
-) {
-    assert_eq!(
-        round_trip_with_options(x, encode_options, decode_options).unwrap(),
-        x
-    )
+    // test non-streaming decompression
+    {
+        let mut bf = std::io::BufReader::new(compressed.as_slice());
+        let mut decomp: Vec<u8> = Vec::new();
+        lzma_rs::lzma_decompress_with_options(&mut bf, &mut decomp, decode_options).unwrap();
+        assert_eq!(decomp, x);
+    }
+
+    #[cfg(feature = "stream")]
+    // test streaming decompression
+    {
+        let mut stream = lzma_rs::decompress::Stream::new_with_options(decode_options, Vec::new());
+
+        if let Err(error) = stream.write_all(&compressed) {
```

Comment:
Why is there an error check here but not in `round_trip_no_options`?

Reply:
You are correct. A simple `.unwrap()` will do. This was because a test case would previously fail with `WriteZero` for some `Options` when `unpacked_size` was reached.

Reply:
I take that back. After running the tests with `--all-features` it shows that `unpacked_size_write_none_to_header_and_use_provided_on_read` can return a `WriteZero` io error:

```
thread 'unpacked_size_write_none_to_header_and_use_provided_on_read' panicked
  at 'called `Result::unwrap()` on an `Err` value:
  Custom { kind: WriteZero, error: "failed to write whole buffer" }', tests/lzma.rs:88:39
```

This is expected behavior. We are encoding `0xFFFF_FFFF_FFFF` in the header but using the provided unpacked size during decoding. Therefore decoding will stop before the 5 to 6 byte end-of-stream marker is read.

The check is here but not in `round_trip_no_options` because it only applies when certain options are provided like in `unpacked_size_write_none_to_header_and_use_provided_on_read`. The `round_trip_no_options` function only uses default options.

I added more documentation and a match statement to make this more clear.